src/trie.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals
+
+
+class Trie(object):
+    """Define class trie."""
+    def __init__(self, token=None, iterable=None):
+        """
+        Create an instance of Trie. <token> must be a string.
+        <iterable> must be a list or a tuple of strings.
+        """
+        self.head = {}
+        if token is not None:
+            self.insert(token)
+        if iterable is not None:
+            if not isinstance(iterable, (list, tuple)):
+                raise TypeError('the iterable must be a list or a tuple.')
+            for token in iterable:
+                self.insert(token)
+
+    def insert(self, token):
+        """
+        Insert the value token into the trie.
+        <token> must be a string.
+        The token is ignored if already present in the trie.
+        """
+        if token and isinstance(type(token), type('word')):
+            raise TypeError('the token must be a string')
+        token = token + '$'
+        current = self.head
+        for char in token:
+            current.setdefault(char, {})
+            current = current[char]
+
+    def contains(self, token):
+        """Return True if token is int trie. False - if notself."""
+        token = token + '$'
+        current = self.head
+        if current is None:
+            return False
+        for char in token:
+            if char in list(current.keys()):
```

Comment:
This call right here is greatly slowing down your `contains` method. Casting the `dict.keys()` to a list is an O(n) operation, plus using the membership operator `in` on a list is another O(n) call.

The good news is that the membership operator on a dict checks the keys and is an O(1) call since dict is a hash table.

Even better, you can use the `try`/`except` structure here.